dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/HostAPIImpl.java

```diff
@@ -787,19 +793,35 @@ private void sendArchiveSiteSystemEvent (final Contentlet contentlet) {
 
     @Override
     @WrapInTransaction
-    public void unarchive(Host host, User user, boolean respectFrontendRoles)
+    public void unarchive(final Host host, final User user, final boolean respectFrontendRoles)
             throws DotDataException, DotSecurityException,
             DotContentletStateException {
-        if(host != null){
+
+        if(host != null) {
+
             hostCache.remove(host);
         }
-        Contentlet c = APILocator.getContentletAPI().find(host.getInode(), user, respectFrontendRoles);
-        APILocator.getContentletAPI().unarchive(c, user, respectFrontendRoles);
+
+        final Contentlet contentlet = APILocator.getContentletAPI()
+                .find(host.getInode(), user, respectFrontendRoles);
+        APILocator.getContentletAPI().unarchive(contentlet, user, respectFrontendRoles);
         host.setModDate(new Date ());
         hostCache.clearAliasCache();
+        HibernateUtil.addAsyncCommitListener(() -> this.sendUnArchiveSiteSystemEvent(contentlet), 1000);
+    }
+
+    private void sendUnArchiveSiteSystemEvent (final Contentlet contentlet) {
+
+        try {
+            
+            DateUtil.sleep(DateUtil.SECOND_MILLIS * 2);
 
```

Comment:
Alternatively, we could use a much modern approach and say something like TimeUnit.SECONDS.sleep(2);

Reply:
it is throwing interrupting exception, the idea is to avoid it